api4/system.go

```diff
@@ -451,6 +457,11 @@ func testS3(c *Context, w http.ResponseWriter, r *http.Request) {
 		cfg = c.App.Config()
```

Comment:
One last thing: let's move the json decoder code here and check for decoding errors.